### ISSUE TITLE
Don't fail on push to master

### ISSUE THREAD

**Issue by matoous** [commitsar-app/commitsar#104]
Currently the action fails on push to master branch.

Commit message:
```
all: Fix tests and lint

Fix failing tests and lint the whole project.
```

Not sure what the desired behavior here is but I would love to use this on my private repo where I push right into the master and repair failed action by ammending to the commit in case anything fails (yup.. not the best practice under the sun).

**Comment by fallion** [comment]
@matoous what would be the expected behaviour here? Because CI will see the last commits as equal. Check last 5 commits?

**Comment by fallion** [comment]
Offending line https://github.com/commitsar-app/commitsar/blob/7cda1dab9463b9d408ca1b7d3a22a1da6e3e844f/cmd/root_runner.go#L61

**Comment by matoous** [comment]
I see a few options here:

1. Lint all commits to last tag.
2. Lint only last commit.
3. Lint X last commits.

I think that combination of 1 and 3 would be best (lint all commits until last tag but cap to 10 commits max maybe?). Either way, printing a warning should be enough to inform the user that this is not the intended use-case and that the behavior might differ from running on separate non-master branch.

**Comment by aexvir** [comment]
I don't think that linting all the commits to the last tag makes, even if capping it makes much sense. 

I can't seem to find numbers, but I'd say that a pretty big amount of PRs usually include a single commit, so, for me, the second options makes more sense. It will end up checking the same, non-related commits many times. And it also can cause issues on projects that didn't follow the convention and just recently they added this step. Also, not a big fan of having an arbitrary number of commits checked, but that might be just me.

Linting just the last commit when there is nothing to compare against, and showing a warning would be the way to go in my opinion.

**Comment by fallion** [comment]
Closed by #145 and #146
